kafka-client/src/main/java/dev/responsive/kafka/internal/db/CassandraFactTable.java

```diff
@@ -190,51 +188,51 @@ public WriterFactory<Bytes> init(
             .build()
     );
 
-    return new FactWriterFactory<>(this);
+    return new FactWriterFactory<>(this, client, kafkaPartition);
   }
 
   @Override
-  public MetadataRow metadata(final int partition) {
+  public long fetchOffset(final int kafkaPartition) {
```

Comment:
👍 I think this change makes sense, probably premature optimization to couple them into a single remote call

Reply:
Mainly I just had noticed that the epoch and offset were never read together/at the same time, so  the optimization  wasn't really doing anything 😞 

That said, I might end up wanting to go back to having a singular `MetadataRow` for the window store where we can persist the stream-time alongside the current offset. But I'll save that for later if/when it's needed